tests/jballerina-unit-test/src/test/resources/test-src/types/intersection/test_intersection_type.bal

```diff
@@ -263,6 +263,68 @@ function testRuntimeTypeNameOfIntersectionType() {
 
 type Error error<record { string message; }>;
 
+type ReadonlyTypeDef1 readonly;
+type ReadonlyTypeDef2 readonly;
+
+
+type ReadonlyObjectType1 object {
+    int atb1;
+    function call() returns int;
+};
+
+type IntersectionType1 ReadonlyTypeDef1 & record {|
+    int a;
+|};
+
+type FooType record {|
+    int a;
+|};
+
+type BarType record {|
+    byte a;
+|};
+
+type IntersectionType2 ReadonlyTypeDef1 & FooType;
+type IntersectionType3 FooType & ReadonlyTypeDef1;
+type IntersectionType4 ReadonlyTypeDef1 & ReadonlyObjectType1;
+type IntersectionType5 ReadonlyObjectType1 & ReadonlyTypeDef1;
+type IntersectionType6 ReadonlyTypeDef1 & ReadonlyTypeDef2 & FooType;
+type IntersectionType7 ReadonlyTypeDef1 & FooType & ReadonlyTypeDef2;
+type ErrIntersection ReadonlyTypeDef1 & error<FooType> & error<BarType>;
+
+function testReadonlyIntersection() {
+    IntersectionType1 foo1 = {a: 1};
+    assertTrue(<any>foo1 is readonly);
+
+    IntersectionType2 foo2 = {a: 1};
+    assertTrue(<any>foo2 is readonly);
+
+    IntersectionType3 foo3 = {a: 1};
+    assertTrue(<any>foo3 is readonly);
+
+    IntersectionType4 obj1 = object {
+        int atb1 = 1;
+        function call() returns int => 1;
+    };
+    assertTrue(<any>obj1 is readonly);
+
+    IntersectionType5 obj2 = object {
+        int atb1 = 1;
+        function call() returns int => 1;
+    };
+    assertTrue(<any>obj2 is readonly);
+
+    IntersectionType6 foo4 = {a: 1};
+    assertTrue(<any>foo4 is readonly);
+
+    IntersectionType7 foo5 = {a: 1};
+    assertTrue(<any>foo5 is readonly);
+
+    ErrIntersection err = error("Error", a=12);
+    assertTrue(<any|error>err is readonly);
+    assertTrue(<any>err.detail().a is byte);
```

Comment:
I would add some tests with both mutable and immutable values in an is check with `T & READONLY`, where READONLY is an alias for readonly.

Reply:
Add a negative test by value too. E.g., trying to use a mutable value where an immutable value is expected.

Reply:
Fixed in [4933a0c55](https://github.com/ballerina-platform/ballerina-lang/pull/42025/commits/4933a0c551b2b9e685b69e459e101a3ce1ee6b5b) and [133d60b769](https://github.com/ballerina-platform/ballerina-lang/pull/42025/commits/133d60b769f21f97dc5aaf477846e5c698ba1999).

Reply:
Meant more like positive and negative runtime tests with `is T & READONLY`

And compile-time error tests like

```ballerina
T x = E;
T & READONLY y = x; // error
```

`

Reply:
added in [commit](https://github.com/ballerina-platform/ballerina-lang/pull/42025/commits/1bff8da100213889d3cabb99a25eb38b6c07f282).

Reply:
> Meant more like positive and negative runtime tests with `is T & READONLY`

Was this added?

Reply:
Added in [commit](https://github.com/ballerina-platform/ballerina-lang/pull/42025/commits/1d0318d4881fc70b972c451da7f6c86c4e6ad136).

Reply:
Still don't seem to have these? The type after `is` has to have the type reference type.

Reply:
Added in [commit](https://github.com/ballerina-platform/ballerina-lang/pull/42025/commits/495a8a56919ce85ea13a6d6dfad1690268d44c09).